cmake/VASTRegisterPlugin.cmake

```diff
@@ -126,23 +134,16 @@ function (VASTRegisterPlugin)
     endif ()
   endif ()
 
-  if (NOT PLUGIN_SOURCES)
-    # Create a stub source file with an identfier if no sources except for the
-    # entrypoint exist.
-    string(MAKE_C_IDENTIFIER "${PLUGIN_TARGET}" plugin_identifier)
-    file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/stub.cpp"
-         "void vast_plugin_${plugin_identifier}_stub() {}")
-    set(PLUGIN_SOURCES "${CMAKE_CURRENT_BINARY_DIR}/stub.cpp")
-  else ()
-    # Deduplicate the entrypoint so plugin authors can grep for sources while
-    # still specifying the entrypoint manually.
-    if ("${PLUGIN_ENTRYPOINT}" IN_LIST PLUGIN_SOURCES)
-      list(REMOVE_ITEM PLUGIN_SPURCES "${PLUGIN_ENTRYPOINT}")
-    endif ()
+  # Deduplicate the entrypoint so plugin authors can grep for sources while
+  # still specifying the entrypoint manually.
+  if ("${PLUGIN_ENTRYPOINT}" IN_LIST PLUGIN_SOURCES)
+    list(REMOVE_ITEM PLUGIN_SPURCES "${PLUGIN_ENTRYPOINT}")
   endif ()
 
-  # Create a static library target for our plugin _without_ the entrypoint.
-  add_library(${PLUGIN_TARGET} STATIC ${PLUGIN_SOURCES})
+  # Create an object library target for our plugin _without_ the entrypoint.
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/stub.h" "")
+  list(APPEND PLUGIN_SOURCES "${CMAKE_CURRENT_BINARY_DIR}/stub.h")
+  add_library(${PLUGIN_TARGET} OBJECT ${PLUGIN_SOURCES})
```

Comment:
nit: It seems that `${PLUGIN_TARGET}-static` is not used in case we're building a shared plugin?

Reply:
It is for the unit test binary.